libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -1743,6 +1743,22 @@ void GCS_MAVLINK::handle_common_gps_message(mavlink_message_t *msg)
     AP::gps().handle_msg(msg);
 }
 
+void GCS_MAVLINK::handle_system_time_message(mavlink_message_t *msg)
+{
+    // exit immediately if system time already set
+    if (hal.util->system_time_set()) {
```

Comment:
This seems a little odd.

Perhaps the clocks are drifting and the GCS wants to bring the clocks back into sync?